src/config/sqs.ts

```
@@ -0,0 +1,105 @@
+		};
+		sqs.sendMessage(params, (err, data) => {
+			if (err) {
+				log.error(err, "Error sending SQS message");
+			} else {
+				log.info(`Successfully added message to sqs queue messageId: ${data.MessageId}`);
+			}
+		});
+	}
+
+	deleteMessage(message: Message) {
+		const deleteParams = {
+			QueueUrl: this.queueUrl,
+			ReceiptHandle: message.ReceiptHandle
+		};
+		sqs.deleteMessage(deleteParams, (err, data) => {
+			if (err) {
+				logger.error({err, data}, "Error deleting message from the queue");
+			} else {
+				logger.debug({data}, "Successfully deleted message from queue");
+			}
+		});
+	}
+
+	executeMessage(message: Message, messageHandler) {
+		const messagePayload = JSON.parse(message.Body);
+		logger.info({messagePayload}, "Sqs message received");
+
+		try {
+			messageHandler({data: messagePayload})
+			this.deleteMessage(message)
+		} catch (err) {
+			logger.error(err, "error executing sqs message")
+		}
+	}
+
+	async handleSqsResponse(err: AWSError, data: ReceiveMessageResult, messageHandler) {
+		if (err) {
+			logger.error({err}, `Error receiving message from SQS queue, queueName ${this.queueName}`);
+		} else {
+			if (!data.Messages) {
+				logger.debug("Nothing to process");
+				return;
+			}
+
+			await Promise.all(data.Messages.map(message => {
+				return new Promise<void>((resolve) => {
+					this.executeMessage(message, messageHandler)
+					resolve()
+				})
+			})
+			)
+		}
+	}
+
+	listen(messageHandler) {
+		// Setup the receiveMessage parameters
+		const params = {
+			QueueUrl: this.queueUrl,
+			MaxNumberOfMessages: 1,
+			VisibilityTimeout: 0,
+			WaitTimeSeconds: 5
+		};
+		sqs.receiveMessage(params, async (err, data) => {
+			try {
+				await this.handleSqsResponse(err, data, messageHandler)
+			} finally {
+				this.listen(messageHandler)
+			}
+		});
+	}
+}
+
+export const sqsQueues: { [key: string]: SqsQueue } = {
+	push: new SqsQueue("push", envVars.SQS_WEBHOOKSQS_QUEUE_URL || "http://localhost:9602/queue/webhooksqs")
```

queue should always use the environment variable as the dev should have the default one in their `.env` file and should fail if it's missing.

---

What change are you suggesting here?